File: frontend/services/calendar.ts
import { api } from './api'

export interface CalendarEventResponse {
  id: string
  type: 'task' | 'project'
  title: string
  due_date: string
  status: string | null
  priority: string | null
  project_name: string | null
  progress: number | null
  completed: boolean | null
}

export const calendarService = {
  /**
   * Load unified calendar deadlines events.
   */
  async getCalendarEvents(params?: {
    start_date?: string
    end_date?: string
  }): Promise<CalendarEventResponse[]> {
    const response = await api.get<CalendarEventResponse[]>('/calendar/events', { params })
    return response.data
  }
}

File: backend/app/schemas/calendar.py
import uuid
from datetime import datetime
from typing import Optional
from pydantic import BaseModel

class CalendarEventResponse(BaseModel):
    id: uuid.UUID
    type: str  # "task" or "project"
    title: str
    due_date: datetime
    status: Optional[str] = None
    priority: Optional[str] = None
    project_name: Optional[str] = None
    progress: Optional[int] = None
    completed: Optional[bool] = None

    class Config:
        from_attributes = True
